### ISSUE TITLE
False positive warning during BiocCheck - Bioconductor software dependencies 

### ISSUE THREAD

**Issue by boyiguo1** [Bioconductor/BiocCheck#207]
Hi, 

Thanks for creating and maintaining the package!

I recently experienced false positive warnings that relates to the check on Bioconductor sofware dependencies during Bioconductor Single Package Building. 

In short, I tried "Imports" or "Depends" on Bioconductor packages, where the check kept prompt warning messages. 
> * Checking for Bioconductor software dependencies...
>     * WARNING: No Bioconductor dependencies detected; consider a CRAN
submission.

See more details via https://github.com/Bioconductor/Contributions/issues/3264#issuecomment-1885564697

I don't see other issues mentioned this problem, and hence report here.

Thanks!

Best,
Boyi

**Comment by LiNk-NY** [comment]
Hi @boyiguo1 
I can't seem to reproduce this locally: 

```r
> BiocCheck:::checkDESCRIPTIONFile("~/bioc/tpSVG/")
* Checking License: for restrictive use...
* Checking for recommended DESCRIPTION fields...
* Checking for Bioconductor software dependencies...
    Bioconductor dependencies found in Imports & Depends (62%).
* Checking for pinned package versions in DESCRIPTION...
```

This suggests that there may be something different in the SPB... I would have to investigate more.

**Comment by boyiguo1** [comment]
Thank you very much!

**Comment by LiNk-NY** [comment]
@boyiguo1 
No problem! Can you bump your package version and push to `git.bioconductor.org` for another build?

**Comment by boyiguo1** [comment]
I just did. The warning message is still there with `BiocCheckVersion: 1.39.16`. Here is the build report (https://bioconductor.org/spb_reports/tpSVG_buildreport_20240112075527.html).

In addition, I saw there's a new error message
> * Checking package installation calls in R code...
Error in .getDirFiles(rowdf["filename"]) : 
  <internal> 'fpaths' input must be a character vector
Calls: BiocCheck ... findSymbolsInParsedCode -> apply -> FUN -> sprintf -> .getDirFiles
Execution halted

I'm not 100% sure if it is caused by updates in v1.39.16. I didn't had the error last build run where there's no significant change except I put a package from "Depends" to "Imports" (https://github.com/boyiguo1/tpSVG/commit/89ee5a39a31fbebce2c309cbf8c86204ea873c1a).

Thank you so much!

**Comment by LiNk-NY** [comment]
Thanks Boyi @boyiguo1 
Please disregard the false positive warning on the SPB. 
I have updated the package to resolve the error. Thanks!

**Comment by LiNk-NY** [comment]
I found the issue. It looks like there is a difference when the check is run on the tarball. 
It should be patched in version `1.39.18`.
